### DISCUSSION TITLE
Confuse with pnpm cache

### DISCUSSION THREAD

**Discussion by SeasonCCC** [pnpm/pnpm#8464] [General]
I'm using TypeORM to connect to a MySQL database, and TypeORM has mysql2 listed as a peer dependency. It imports mysql2 using [require("mysql2")](https://github.com/typeorm/typeorm/blob/master/src/platform/PlatformTools.ts). When I use pnpm install typeorm mysql2, everything works fine. However, when I use pnpm remove mysql2, the .pnpm directory inside node_modules still keeps the mysql2 dependency. This results in a situation where even though I've removed the mysql2 dependency, TypeORM can still find it normally, and the database connection remains functional.

**Comment by zkochan** [comment]
I am not sure why typeorm can find it. pnpm should remove all the symlinks pointing to it. If you want to disable this case, set [modules-cache-max-age=0](https://pnpm.io/npmrc#modules-cache-max-age)

**Reply by SeasonCCC** [reply]
I find that the problem comes from the cross-env package. When i use `cross-env NODE_ENV=development node index.js` to run nodejs file, dependencies inside .pnpm directory can be found it normally.
